java/code/src/com/redhat/rhn/manager/channel/CloneChannelCommand.java

```diff
@@ -117,14 +116,9 @@ public Channel create()
         ChannelFactory.save(c);
         c = ChannelFactory.reload(c);
 
+        // Comps files are cloned by the DB trigger 'rhn_channel_cloned_comps_trig'
         if (stripModularMetadata) {
-            if (c.getModules() != null) {
-                HibernateFactory.getSession().delete(c.getModules());
-            }
-            c.setModules(null);
-        }
-        else {
-            c.cloneModulesFrom(original);
+            c.getModules().clear();
```

Comment:
I think getModules can return null. It can be tested with:
```suggestion
            if (c..isModular()) {
                c.getModules().clear();
             }
```